sklearn/tree/_splitter.pxd

```diff
@@ -27,6 +27,8 @@ cdef struct SplitRecord:
     double improvement     # Impurity improvement given parent node.
     double impurity_left   # Impurity of the left split.
     double impurity_right  # Impurity of the right split.
+    unsigned char missing_go_to_left  # Missing value go left
```

Comment:
`bool` is not an actual C type. The closest alternative is `bint`. I used `unsigned char` to be consistent with hist gradient boosting:

https://github.com/scikit-learn/scikit-learn/blob/44860bd5923475e769fa3addb03d4578410266c5/sklearn/ensemble/_hist_gradient_boosting/splitting.pyx#L34

`unsigned char` was used there because it is the same as `np.bool_`: https://github.com/numpy/numpy/blob/3b8dca3d0588048c425488a5e39ceddf815d9138/numpy/__init__.cython-30.pxd#L292